context/secrets.js

```diff
@@ -17,9 +18,17 @@ export function useSecretsContext() {
 export function SecretsProvider({ children }) {
   const [secrets, setSecrets] = useState([])
 
+  useEffect(() => {
+    ;(async () => {
+      const secretsManager = await SecretsManager.getInstance()
```

Comment:
let's avoid singletons. if we want a single instance of this object let's create it once and reuse it but avoid this getInstance pattern

Reply:
Fair enough! I'll make the adjustments.